### ISSUE TITLE
Blockchain wallet - blockchain2john.py script error - No JSON object could be decoded

### ISSUE THREAD

**Issue by kujina** [openwall/john#4544]
Windows 10 64bit
Python 2.1.18

Yesterday I downloaded a Blockchain.com wallet that I created at Blockchain.info in **December 2012**. I downloaded the wallet.aes.json wallet using the btcrecover download-blockchain-wallet.py script.

I've then tried to get a hash from this wallet using the blockchain2john.py script but I get the following error.

C:\john-bleeding-jumbo\run>C:\python27\python blockchain2john-copy.py --json wallet.aes.json >> blockchainhash.txt
Traceback (most recent call last):
  File "blockchain2john-copy.py", line 42, in <module>
    decoded_data = json.loads(data.decode("utf-8"))
  File "C:\python27\lib\json\__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "C:\python27\lib\json\decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\python27\lib\json\decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded

I'm not sure if this is because of the age/type of wallet or that for some reason my wallet.aes.json is bad.

**Comment by solardiz** [comment]
Hi @kujina. What you posted reads more like a user support request (off-topic here, would be OK on john-users mailing list) than an issue with our project. That said, let's discuss it a bit to figure out whether there's an issue we should look into or not.

What is the size of your `wallet.aes.json` file? Can you manually check whether this file contains any JSON data at all? https://en.wikipedia.org/wiki/JSON#Syntax

What Python version did you use to run `download-blockchain-wallet.py`? IIRC, last time I checked it needed at least Python 3.6, whereas above you show you're using Python 2.7. I have no idea what "Python 2.1.18" means - possibly some version of a Windows build of Python, rather than of Python itself?

Basically, you need to check whether your use of `download-blockchain-wallet.py` worked as intended or not before it'd make sense to look into a possible issue with `blockchain2john.py`.

BTW, why is it called `blockchain2john-copy.py` on your system? Is it any different from what we have in this repository?

**Comment by kujina** [comment]
Apologies my mistake I was up very late here yesterday. My Blockchain wallet.aes.json file is 344 bytes and is **not** in JSON formatting, I ran `blockchain2john-copy.py wallet.aes.json >> blockchainhash.txt` without the `--json`  and got an output. You can delete/close this Issue, thanks.
